### DISCUSSION TITLE
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?

### DISCUSSION THREAD

**Discussion by lidongze0629** [rancher-sandbox/rancher-desktop#7291] [Q&A]
Rancher Desktop Version: 1.15.0-arm64
MacBook Pro: Ventura 13.4.1 (Apple M2 Pro)

After following this doc(https://docs.docker.com/desktop/uninstall/), I installed Rancher desktop. There is no report in diagnostics, however, I still can‘t use `docker` common

<img width="1196" alt="截屏2024-08-01 17 53 23" src="https://github.com/user-attachments/assets/dbc9b9f1-8db8-48c2-a619-166e974d3c50">

```shell
$ docker images
Cannot connect to the Docker daemon at unix:///var/run/docker.sock. Is the docker daemon running?
```

### Some attempts
- I tried to exec `sudo ln -s ~$USER/.rd/docker.sock /var/run/docker.sock` from [here](https://github.com/rancher-sandbox/rancher-desktop/issues/2534), but it didn't work for me:

```shell
$ sudo ln -s ~$USER/.rd/docker.sock /var/run/docker.sock
$ docker images
error during connect: Head "http://%2Fvar%2Frun%2Fdocker.sock/_ping": EOF
```

- I tried to exec `docker context use rancher-desktop`, but got context not found problem

```
$ docker context use rancher-desktop
context "rancher-deskto": context not found: open /Users/xxxxx/.docker/contexts/meta/16732ca79f7c4242394423dfde8826f62b3e8ccafc5b828d83ed42b8138c8f72/meta.json: no such file or directory
```

Does somebody face the same problem? Thanks very much for giving me any advice !

**Comment by jandubois** [comment]
Are you sure you configured Rancher Desktop to use the `moby` container engine, and not `containerd`? You cut of the status line in your screenshot, so I can't tell.

**Reply by lidongze0629** [reply]
<img width="775" alt="截屏2024-08-02 17 12 18" src="https://github.com/user-attachments/assets/128b2f2b-9428-413c-bd52-d556e24c7ca8">

Thanks for your feedback. Here is my preferences page.

**Comment by mook-as** [comment]
> context "rancher-deskto": context not found: open /Users/xxxxx/.docker/contexts/meta/16732ca79f7c4242394423dfde8826f62b3e8ccafc5b828d83ed42b8138c8f72/meta.json: no such file or directory

That hash is incorrect; we're expecting `b547d66a5de60e5f0843aba28283a8875c2ad72e99ba076060ef9ec7c09917c8` instead:
https://github.com/rancher-sandbox/rancher-desktop/blob/0901d499281b09df7603c8db076a7ae53dc83a4f/pkg/rancher-desktop/utils/dockerDirManager.ts#L62-L63

```
$ echo -n rancher-desktop | sha256sum 
b547d66a5de60e5f0843aba28283a8875c2ad72e99ba076060ef9ec7c09917c8  -
$ echo -n rancher-deskto | sha256sum 
16732ca79f7c4242394423dfde8826f62b3e8ccafc5b828d83ed42b8138c8f72  -
```

Did you miss the `p` in `rancher-desktop`?

**Reply by lidongze0629** [reply]
@mook-as Thanks, I did miss the `p`. However, I don't have any context in this directory `~/.docker/`

```bash
$ docker context use rancher-desktop
context "rancher-desktop": context not found: open /Users/lidongze/.docker/contexts/meta/b547d66a5de60e5f0843aba28283a8875c2ad72e99ba076060ef9ec7c09917c8/meta.json: no such file or directory


$ cd ~/.docker && ls
cli-plugins config.json

$ cat config.json
{
  "credsStore": "osxkeychain"
}
```

**Comment by kurianm** [comment]
I am also facing the same issue. How can we solve the problem

**Comment by jandubois** [comment]
If you are sure that you have selected the `moby` container engine, but don't get the `rancher-desktop` docker context, then something must have failed before the context was created.

Please make sure the problem can be reproduced with the 1.15.1 release; there was a bug in an earlier versions that could have caused this problem, that has been fixed in 1.15.1.

So if you can reproduce the problem with 1.15.1 then please enable "Debug" logging in the [Troubleshooting page](https://docs.rancherdesktop.io/ui/troubleshooting), restart Rancher Desktop one more time, and then create a new GitHub issue, fill out all the details about your setup, and attach a ZIP file of all the log files.
